### ISSUE TITLE
Add the MSAL4J library to enable Azure AD (Entra) authentication with Microsoft identities

### ISSUE THREAD

**Issue by rplescia** [metabase/metabase#37493]
**Is your feature request related to a problem? Please describe.**
When trying to add an SQL Server (Azure SQL) database hosted in Microsoft Azure, using an Azure AD account for authentication and the 'authentication=ActiveDirectoryPassword' JDBC connection property, it fails due to this library being required for Azure AD authentication.

**Describe the solution you'd like**
Package the MSAL4J library with the next release of Matabase.

**Describe alternatives you've considered**
I'm currently running a PoC environment in Docker, I have tried to deploy the library to the running container to see if it will fix my issue but I've had much luck altering the container. As we would probably run it in a container in production, it would be great if this library was included.

**How important is this feature to you?**
Very. We are a Microsoft house and we are in the process of migrating to Azure, so we need to connect to our data sources with Azure AD

**Additional context**
[https://discourse.metabase.com/t/unable-to-connect-to-azure-synapse-sql-database/56325](<url>)

**Comment by niclas-evidence-operations** [comment]
We are also doing a PoC with Azure databases and would require Entra ID based authentication. Is that in the plan to support?

**Comment by paoliniluis** [comment]
It's not on the short term plans, unless this is something that an enterprise customer requires

**Comment by gitdrk** [comment]
Adding a note. I have been working on a metabase POC and cannot connect to Fabric/Azure Synapse due to this.

**Comment by paoliniluis** [comment]
@rplescia I guess that this is not the only requirement to authenticate to a service with Entra, am I correct?
https://learn.microsoft.com/en-us/fabric/data-warehouse/connectivity#connect-using-jdbc

**Comment by gitdrk** [comment]
@paoliniluis - Azure continues to push their services to Entra Id. As this continues it is going to become difficult for Azure centered organizations like mine to adopt Metabase. I had a conversation with MS the other day about a potential pivot from Synapse -> Fabric and Entra Id is unconditional for connecting to the warehouse.

Are you aware of any workaround for supporting Fabric and other Azure data stores? We are hoping to continue using Metabase internally as well as for customer facing embedded analytics.

**Comment by paoliniluis** [comment]
@gitdrk Microsoft being Microsoft. We'll try our best here

**Comment by gitdrk** [comment]
Thanks @paoliniluis ! Feel free to reach out to me if I can be of assistance in explaining our stack!

**Comment by martintburgess** [comment]
Plus one - I'd love to use Metabase at my company but Entra ID is a non-negotiable from Microsoft/IT.

**Comment by ajdickerson83** [comment]
I would love to try this but REALLY need Entra to be supported.

**Comment by radiance-vouk** [comment]
Hi - is there a status on getting Fabric connectivity using authentication=ActiveDirectoryServicePrincipal.  There are no viable workarounds.  Microsoft customers are all enterprise - really suprised this is not supported yet - considering features are only added if an enteprise customer requires.

**Comment by dragonsahead** [comment]
@radiance-vouk are you an enterprise customer? if so, please contact us

**Comment by martintburgess** [comment]
I imagine it's a bit of a chicken-egg issue. It would be hard for any Microsoft organisations to use metabase because the functionality isn't it in place. The functionality won't get in place until they are a metabase customer..

**Comment by dragonsahead** [comment]
@martintburgess Microsoft companies tend to use PowerBI since it's free for them, so unless we have a really good reason to support them we focus our efforts somewhere else

**Comment by goto-loop** [comment]
Let's be real here: Microsoft companies tend to use PowerBI because they're stuck with it, and right now they can't even try Metabase to see what analytics can actually be like. I'm not sure how complex the implementation of Entra authentication would be, but it seems like a minor issue compared to missing out on the whole Microsoft crowd?

**Comment by thomasforde** [comment]
I totally agree, i was looking at this solution now because i dont like PowerBI to work with, its heavy, not good visuals, and definitly not free as i want to share dashboards with my team, and it caught me by suprise that entraid was not supported out of the box. Everybody uses entraid these days.. 

I would say in our instracture, we set EntraID as a criteria because we can use cloud security systems to control access. Now i have to look elsewhere for something similar.

**Comment by Shashwat-pixel** [comment]
Is there any update to this? I have been trying to connect Microsoft Fabric to Metabase as well. Tried different drivers within Metabase plugins, but to no avail..
Will this functionality be available to Metabase any time soon? 
Thank you. Much appreciated..

**Comment by GlebKaplan** [comment]
Count my vote in please. We have all operational sources available in Metabase but Dataverse.

**Comment by dragonsahead** [comment]
@GlebKaplan @Shashwat-pixel @thomasforde @goto-loop @radiance-vouk @martintburgess @gitdrk @rplescia @niclas-evidence-operations can someone test this PR and let us know if it made any differece? https://github.com/metabase/metabase/pull/82511
